numpy/core/src/multiarray/iterators.c

```diff
@@ -1266,13 +1276,14 @@ static PyObject*
 multiiter_new_impl(int n_args, PyObject **args)
 {
     PyArrayMultiIterObject *multi;
+    npy_bool needs_gc_tracking = NPY_FALSE;
     int i;
 
-    multi = PyArray_malloc(sizeof(PyArrayMultiIterObject));
+    multi = ((PyArrayMultiIterObject *)PyArrayMultiIter_Type.
+            tp_alloc(&PyArrayMultiIter_Type, 0));
```

Comment:
Ideally you would pass through `subtype` from `arraymultiter_new`, and use it in place of `PyArrayMultiIter_Type` here